src/librustc_trans/mir/block.rs

```diff
@@ -116,6 +116,9 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                 if let Some(cleanup_pad) = cleanup_pad {
                     bcx.cleanup_ret(cleanup_pad, None);
                 } else {
+                    let llpersonality = bcx.fcx().eh_personality();
+                    bcx.set_personality_fn(llpersonality);
```

Comment:
I believe I said already, this code seems dubiously workaround-ish. Can we fix this properly, by fixing (if its not enough) up in the function near line 80.